api/v1alpha1/workflow_webhook.go

```diff
@@ -181,25 +190,27 @@ func (w *Workflow) ValidateDelete() error {
 	return nil
 }
 
-func validateWorkflowImmutable(a *Workflow, b *Workflow) error {
-	if a.Spec.WLMID != b.Spec.WLMID {
-		return fmt.Errorf("cannot change immutable field WLMID")
+func validateWorkflowImmutable(wfNew *Workflow, wfOld *Workflow) error {
+
+	specPath := field.NewPath("Spec")
+	if wfNew.Spec.WLMID != wfOld.Spec.WLMID {
+		return field.Forbidden(specPath.Child("WLMID"), "cannot change immutable field WLMID")
```

Comment:
Is providing the field, WLMID, and the message "cannot change immutable field WLMID" redundant when the error message is "Spec.WLMID: Forbidden: Cannot change immutable field WLMID"

Reply:
maybe a helper function?
```go
immutableError := func(field string) error {
   return field.Forbidden(field.NewPath("Spec").Child(field)), "cannot change immutable field")
}

if old.Spec.Field != new.Spec.Field {
   return immutableError("Field")
}
```